src/storage/DataAccessorBasedStore.ts

```diff
@@ -98,7 +92,7 @@ export class DataAccessorBasedStore implements ResourceStore {
 
     // When a POST method request targets a non-container resource without an existing representation,
     // the server MUST respond with the 404 status code.
-    if (!parentMetadata && !container.path.endsWith('/')) {
+    if (!parentMetadata && !isContainerPath(container.path)) {
```

Comment:
```suggestion
    if (!parentMetadata && !isContainer(container)) {
```

such that the caller does not need to know how how the callee determines container-ness?

Reply:
There is 1 situation though where we also use it to check if the `slug` ends with a slash to determine if we need to create a container or a document. That's the reason I chose this.

Reply:
Maybe expose both?